PSKoans/Data/Advice/Fun/FunWithCats.Advice.json

```
@@ -5,20 +5,17 @@
     "\n",
     "It can also be fun!",
     "\n",
-    "We've provided some fun examples below just to give you a taste of what PowerShell can do",
+    "We've provided some fun examples below just to give you a taste of what PowerShell can do.",
+    "\n",    
     "Cats",
     "\n",
-    "   The below code snippit will query a RESTAPI on the internet that provides some fun facts about cats. It then uses the SpeechSynthesizer to tell you these facts! Feel free to run it on your machine:",
+    "   The below code snippet will query a REST API on the internet that provides some fun facts about cats. It then uses text-to-speech to tell you these facts! Feel free to run it on your machine:",
     "\n",
     "function Get-CatFact {",
-    "\n",
-    "\tAdd-Type -AssemblyName System.Speech",
-    "\t$SpeechSynth = New-Object System.Speech.Synthesis.SpeechSynthesizer",
-    "\t$URI = 'http://www.catfact.info/api/v1/facts.json?per_page=200'",
-    "\t$Random = Get-Random -Maximum 200",
-    "\t$CatFact = (ConvertFrom-Json (Invoke-WebRequest -Uri $URI)).Facts[$Random]",
-    "\t$SpeechSynth.Speak('did you know?')",
-    "\t$SpeechSynth.Speak($CatFact.details)",
+    "\t$SpeechSynth = New-Object -ComObject SAPI.SpVoice",
+    "\t$URI = 'https://catfact.ninja/fact'",
+    "\t$CatFact = Invoke-WebRequest -Uri $URI | ConvertFrom-Json | Select-Object -ExpandProperty fact",
```

```suggestion
    "\t$CatFact = Invoke-RestMethod -Uri $URI | Select-Object -ExpandProperty fact",
```

IRM does a direct JSON conversion for this purpose 🙂

---

And done. Thanks 🙂